### PR TITLE
Add plugin for figure caption

### PR DESCRIPTION
**Pull request by bebatut**
⚠️  Depends on #432 

---

Hi,

@willdurand developed a plugin to automatically add caption to figure using either the `title` or the `alt` information in the markdown figure (`alt`, with a preference for the `title`).
I added the description to use it in the tutorial for writing tutorials 😛 

Bérénice

### CODE REVIEW

**Comment by shiltemann** [comment]
cool beans! this is really nice, thanks both of you :)

File: _plugins/jekyll-figurify.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  class Figurify < Jekyll::Generator
+
+    safe true
+
+    def initialize(config)
+      @config = config['figurify'] ||= {}
+    end
+
+    def generate(site)
+      puts "\n      Applying figurify plugin ✨"
```
Comment (willdurand) [inline _plugins/jekyll-figurify.rb]: @bebatut you should get rid of this line. It was fun but now it makes me crazy and I cannot see it anymore 😛

Comment (bebatut) [inline _plugins/jekyll-figurify.rb]: 😄  I removed it

**Comment by willdurand** [comment]
For the record, the plugin parses the markdown content and transforms markdown images into:

```
<figure id="figure-999">
  <img src="url" alt="alt" title="title">
  <figcation>Figure 999: title</figcaption>
</figure>
```

The plugin does that before the HTML transformation (so after having applied the plugin, you get some HTML into your markdown, which is perfectly valid but still worth mentioning here).

**Pull request merged**